src/components/layout/dashboard/sideNav/useItemSelectionStatus.tsx

```diff
@@ -16,9 +17,12 @@ export const useItemSelectionStatus = <T extends object>({
 
   const isSelected = useMemo(() => {
     if (!item.path && !item.activePaths) return false;
-    // If current location starts with item path, then this nav item should be selected.
+    // If current location matches item path, then this nav item should be selected.
     // (e.g. item path is /enrollments/1/services and current location is /enrollments/1/services/2)
-    if (item.path && currentLocation.startsWith(item.path)) {
+    // Use a regex to avoid false positives (e.g. item path is /projects/1/ce and current location is /projects/1/ce-participation)
+    const escapedPath = item.path ? escapeRegExp(item.path) : '';
+    const regex = new RegExp(`^${escapedPath}(?:[#/\?]|$)`);
+    if (item.path && regex.test(currentLocation)) {
       return true;
```

Comment:
Good find! I'm unsure we need the complexity of the regex. There should always be a `/` between the "base" path and the rest of the path. I _think_ we could resolve the bug of comparing `/projects/1/ce` and `/projects/1/ce-participation` could be resolved by ensuring that the path starts with `item.path + '/'`?

Does this achieve the same result? Maybe there is something I am missing about hashes.
```jsx
    // move to util
    const addTrailingSlash = (path: string) =>
      path.endsWith('/') ? path : path + '/';

    const matchesPath =
      currentLocation === item.path || // matches path exactly
      currentLocation.startsWith(addTrailingSlash(item.path)); // starts with path

    if (matchesPath) return true;
```

Reply:
Thanks! Yes I think it achieves the same result, even for pages with hash params, since `currentLocation` returns just the pathname, not the hash.